homeassistant/components/vizio/media_player.py

```diff
@@ -83,18 +107,32 @@
             "is valid and available, device type is correct%s",
             fail_auth_msg,
         )
-        return
+        raise PlatformNotReady
+
+    entity = VizioDevice(
+        hass, config_entry, device, name, volume_step, device_type, unique_id
+    )
 
-    async_add_entities([VizioDevice(device, name, volume_step, device_type)], True)
+    async_add_entities([entity], True)
 
 
 class VizioDevice(MediaPlayerDevice):
     """Media Player implementation which performs REST requests to device."""
 
     def __init__(
-        self, device: VizioAsync, name: str, volume_step: int, device_type: str
+        self,
+        hass: HomeAssistantType,
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.